src/lib/optimizer/optimizer.hpp

```diff
@@ -10,6 +10,11 @@
 
 namespace opossum {
 
+struct OptimizerRuleMetrics {
+  std::string name;
+  std::chrono::nanoseconds duration;
```

Comment:
Is this overwritten when a rule is executed twice (e.g., PredicateSplitUpRule)?

Reply:
no. a new instance is added to the vector of metrics. later in the python script, multiple runtimes for one rule are added up.

Reply:
Did you test this? The following looks like you would override a value in a dictionary:

```
for (auto rule_duration : sql_statement_metrics->optimizer_rule_durations) {
  rule_metrics_json[rule_duration.name] = rule_duration.duration.count();
}
```

Reply:
If, instead of using the rule's name as the key, we kept a list of applied rules, we could also display the rules in the order the optimizer executed them. However, we need to keep in mind that the upcoming changes to the plan cache might mean that the second execution of a query executes different rules, potentially even in a different order. I have no good solution to that yet.

Reply:
you were right, in the json export, the duration was overwritten. now it's appended to a list and summed up in the python script